src/config.rs

```diff
@@ -326,6 +372,40 @@ fn parse_args() -> String {
     cli_args[1..].join(" ")
 }
 
+fn parse_headers(prefix: &str) -> Result<MetadataVector, Error> {
+    let mut metadata: MetadataVector = MetadataVector { vec: Vec::new() };
+
+    for (key, value) in env::vars() {
+        let stripped_key: Option<&str> = key.strip_prefix(prefix);
+        match stripped_key {
+            Some(stripped_key) => {
+                // attempt to parse the stripped key
+                match AsciiMetadataKey::from_str(stripped_key) {
+                    Ok(metadata_key) => {
+                        // attempt to parse the value
+                        match AsciiMetadataValue::from_str(&value) {
```

Comment:
nit: more idiomatic would probably be to do `AsciiMetadataValue::from_str(&value).ok_or_else(||Error::InvalidHeaderValue(value))?`

Reply:
https://github.com/istio/ztunnel/pull/1460